src/components/ha-button-toggle-group.ts

```diff
@@ -33,6 +37,11 @@ export class HaButtonToggleGroup extends LitElement {
                 <ha-svg-icon .path=${button.iconPath}></ha-svg-icon>
               </mwc-icon-button>`
             : html`<mwc-button
+                style=${ifDefined(
+                  this.fullWidth
+                    ? styleMap({ width: `${100 / this.buttons.length}%` })
+                    : undefined
+                )}
```

Comment:
Can't you make the style map conditional? 

```suggestion
                style=${styleMap({ width: this.fullWidth ? 'initial' : `${100 / this.buttons.length}%` })}
```